Client/Frontend/Home/ActivityStreamPanel.swift

```diff
@@ -611,7 +611,7 @@ extension ActivityStreamPanel: DataObserverDelegate {
     }
 
     // Invoked by the ActivityStreamDataObserver when highlights/top sites invalidation is complete.
-    func didInvalidateDataSources(forceHighlights highlights: Bool, forceTopSites topSites: Bool) {
+    func didInvalidateDataSources(forceHighlights highlights: Bool, forceTopSites topSites: Bool, highlightsRefreshed: Bool, topSitesRefreshed: Bool) {
```

Comment:
didInvalidateDataSources/willInvalidate don't need `forceHighlights` `forceTopSites` We can just replace them with the new params.

Reply:
I think if we change `forceHighlights` to `highlightsRefreshed` that will cause additional calls to `ActivityStreamPanel.reloadAll` because there are cases where `highlightsRefreshed` is true but `forceHighlights` is false. 

https://github.com/mozilla-mobile/firefox-ios/blob/master/Client/Frontend/Home/ActivityStreamPanel.swift#L611

Reply:
Hmm. I see what you mean. My bad, I see what I was trying to do here 😆 . I feel that the `didInvalidateDataSources` method is way too long. How about 
`func didInvalidateDataSources(refresh forced: Bool, highlightsRefreshed: Bool, topSitesRefreshed: Bool)`
where forced is simply `forceHighlights || forceTopSites`

Reply:
Sounds good. I'll update the PR with that change.